crates/script/src/build.rs

```diff
@@ -48,9 +47,198 @@ impl BuildData {
         Linker::new(self.project_root.clone(), self.output.artifact_ids().collect())
     }
 
+    fn get_zk_linker(&self, script_config: &ScriptConfig) -> Result<ZkLinker<'_>> {
+        let zksolc = foundry_config::zksync::config_zksolc_compiler(&script_config.config)
+            .context("retrieving zksolc compiler to be used for linking")?;
+        let version = zksolc.version().context("trying to determine zksolc version")?;
+
+        let Some(input) = self.zk_output.as_ref() else {
+            eyre::bail!("unable to link zk artifacts if no zk compilation output is provided")
+        };
+
+        let linker =
+            ZkLinker::new(self.project_root.clone(), input.artifact_ids().collect(), zksolc, input);
+
+        let mut libs = Default::default();
+        linker.zk_collect_dependencies(&self.target, &mut libs, None)?;
+
+        // if there are no no libs, no linking will happen
+        // so we can skip version check
+        if !libs.is_empty() && version < DEPLOY_TIME_LINKING_ZKSOLC_MIN_VERSION {
+            eyre::bail!(
+                "deploy-time linking not supported. minimum: {}, given: {}",
+                DEPLOY_TIME_LINKING_ZKSOLC_MIN_VERSION,
+                &version
+            );
+        }
+
+        Ok(linker)
+    }
+
+    /// Will attempt linking via `zksolc`
+    ///
+    /// Will attempt linking with a CREATE2 deployer if possible first, otherwise
+    /// just using CREATE.
+    /// After linking is done it will update the list of `DualCompiledContracts` with
+    /// the newly linked contracts (and their EVM equivalent).
+    /// Finally, return the list of known contracts
+    ///
+    /// If compilation for zksync is not enabled will return the
+    /// given EVM linked artifacts
+    async fn zk_link(
+        &mut self,
+        script_config: &ScriptConfig,
+        known_libraries: Libraries,
+        evm_linked_contracts: ArtifactContracts,
+        use_create2: bool,
+    ) -> Result<ArtifactContracts> {
+        if !script_config.config.zksync.should_compile() {
+            return Ok(evm_linked_contracts);
+        }
+
+        let Some(input) = self.zk_output.as_ref() else {
+            eyre::bail!("unable to link zk artifacts if no zk compilation output is provided");
+        };
+
+        let mut dual_compiled_contracts = self.dual_compiled_contracts.take().unwrap_or_default();
```

Comment:
How do we have access to `dual_compiled_contracts` here, thought we moved them to strategy.

Reply:
Script doesn't use strategies